lib/ecstatic.js

```diff
@@ -330,6 +339,24 @@ var ecstatic = module.exports = function (dir, options) {
 ecstatic.version = version;
 ecstatic.showDir = showDir;
 
+function hasGzipId12(gzipped, cb) {
+  var stream = fs.createReadStream(gzipped, { start: 0, end: 1 }),
+      buffer = Buffer(''),
+      error = null;
+
+  stream.on("error", function (err) {
+    error = err;
+  });
+
+  stream.on("data", function (chunk) {
+    buffer = Buffer.concat([buffer, chunk], 2);
+  });
+
+  stream.on("close", function () {
```

Comment:
Docs say close is not called when error is emitted. You have to cb separately in the error case.